libGeoIP/GeoIP.c

```diff
@@ -1499,29 +1492,27 @@ GeoIP * GeoIP_open_type(int type, int flags)
 GeoIP * GeoIP_new(int flags)
 {
     GeoIP * gi;
-    _GeoIP_setup_dbfilename();
-    gi = GeoIP_open(GeoIPDBFileName[GEOIP_COUNTRY_EDITION], flags);
+    gi = GeoIP_open(_GeoIP_get_dbfilename(GEOIP_COUNTRY_EDITION), flags);
     return gi;
 }
 
 GeoIP * GeoIP_open(const char * filename, int flags)
 {
+/*
+        Ownership of filename is transferred to the GeoIP object
+ */
```

Comment:
This seems fine for _internal_ uses, but I don't think we want to change the behavior generally given that this is a public function and that is probably more used than `GeoIP_new`.